core_v2.go

```diff
@@ -58,6 +61,14 @@ func newInstantiateOptions(opts ...InstantiateOption) instantiateOptions {
 		WithSwitcherServerAutoStart().apply(&options)
 	}
 
+	// Enable auto-reset by default in production environments with 30 minute timeout
+	if options.logLevelAutoResetEnabled == nil && options.isProductionEnvironment() {
+		WithLogLevelAutoResetEnabled().apply(&options)
+	}
```

Comment:
The options name should be:
- logLevelSwitcherServerAutoResetEnabled
- logLevelSwitcherServerAutoResetTimeout

And update the corresponding `With...` version.